src/simoc_sam/sensors/basesensor.py

```diff
@@ -186,8 +186,11 @@ def print(self, *args, **kwargs):
 
     def log(self, payload):
         # TODO: implement better logging
-        with open(self.log_fname, 'a') as f:
-            f.write(f'{payload}\n')
+        try:
+            with open(self.log_fname, 'a') as f:
+                f.write(f'{payload}\n')
+        except FileNotFoundError as err:
```

Comment:
```suggestion
        except Exception as err:
```
Better to use a more generic exception.